webboot/webboot.go

```diff
@@ -212,9 +212,17 @@ func main() {
 			log.Fatal(err)
 		}
 	}
-
+	//the Request function sets up a DHCP confifuration for all interfaces,
+	//such as eth0, which is an ethernet interface.
 	if *ipv4 || *ipv6 {
+		//ifname uses regex to find patterns.
+		//it uses ^ as the start of the line,
+		//[we] to find any single character containing w,e
+		//. to find any single charachter
+		//and * suggesting 0 or more or ^[we].
```

Comment:
almost!!!
^ - start of line
[we] --   either one
. -- any character
* -- modifies the ., not the [we]

here's how to play with regex
grep '^[we].*'
then start typing typical names and see what happens
e.g.
grep '^[we].*'
eth0
wlan09
alan0

and so on

you're close

Reply:
i haven't tried it (not home).... but i'm guessing the `w` in `^[we].*` is for wlan0 and `e` is eth0... just a hunch.

Reply:
which is true, but your description of the * is wrong. the . and * and what they mean are supremely important ...

Reply:
. is any character 
* is expect more than ^[we] 

(this is what https://yourbasic.org/golang/regexp-cheat-sheet/ says)